python/tank/deploy/tank_commands/core_upgrade.py

```diff
@@ -251,184 +256,133 @@ def __init__(self, installation_root, logger, core_version=None):
                                   e.g. you can run the updater as a totally separate thing.
         :param logger: Logger to send output to.
         :param core_version: Version of the core to update to. If None, the core will be updated to the latest
-                             version.
+                             version. Defaults to None.
         """
         self._log = logger
 
-        (sg_app_store, script_user) = shotgun.create_sg_app_store_connection()
-        self._sg = sg_app_store
-        self._sg_script_user = script_user
+        from tank_vendor.shotgun_deploy import Descriptor, create_descriptor
 
-        self._local_sg = shotgun.get_sg_connection()
-        self._update_version = self.__get_core_version(core_version)
+        self._current_core_descriptor = create_descriptor(
```

Comment:
I think it might be better to add a `get_latest_core_descriptor` helper method to the pipeline config object, and basically mimic the methods that are already there for apps, engines, and frameworks. This would better take into account stuff like bundle override paths - something like:

```python
    def get_latest_core_descriptor(self):
        """
        Convenience method that returns the latest descriptor for core

        :returns:        Descriptor object
        """
        return self._get_descriptor(Descriptor.CORE, "sgtk:descriptor:app_store?name=tk-core", latest=True)
```